llvm/lib/Target/AIE/AIETargetTransformInfo.h

```diff
@@ -62,6 +62,15 @@ class AIETTIImpl : public BasicTTIImplBase<AIETTIImpl> {
     UP.Threshold = 200;
     BaseT::getUnrollingPreferences(L, SE, UP, ORE);
   }
+
+  bool addrspacesMayAlias(unsigned AS0, unsigned AS1) const {
+    if (AS0 == AS1)
+      return true;
+
+    const AIEBaseAddrSpaceInfo &ASI = ST->getAddrSpaceInfo();
```

Comment:
I think that we need some refactor here. we have almost the same code. If we can delegate this to `AIEBaseAddrSpaceInfo` then we can eliminate this double implementation. We just need to reimplement in AIEBaseAddrSpaceInfo.